internal/services/cdn/cdn_frontdoor_route_disable_link_to_default_domain_resource.go

```diff
@@ -163,143 +164,173 @@ func resourceCdnFrontDoorRouteDisableLinkToDefaultDomainRead(d *pluginsdk.Resour
 	defer routeCancel()
 
 	customDomainClient := meta.(*clients.Client).Cdn.FrontDoorCustomDomainsClient
-	customDomainCtx, customDomaincancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
-	defer customDomaincancel()
+	customDomainCtx, customDomainCancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer customDomainCancel()
+
+	id, err := parse.FrontDoorRouteDisableLinkToDefaultDomainID(d.Id())
+	if err != nil {
+		return err
+	}
 
 	routeId, err := parse.FrontDoorRouteID(d.Get("cdn_frontdoor_route_id").(string))
 	if err != nil {
-		return fmt.Errorf("unable to parse CDN Front Door Route ID: %+v", err)
+		return fmt.Errorf("front door route disable link to default domain: %+v", err)
 	}
 
 	// Make sure the route still exist...
-	routeResp, err := routeClient.Get(routeCtx, routeId.ResourceGroup, routeId.ProfileName, routeId.AfdEndpointName, routeId.RouteName)
+	existing, err := routeClient.Get(routeCtx, routeId.ResourceGroup, routeId.ProfileName, routeId.AfdEndpointName, routeId.RouteName)
 	if err != nil {
+		if utils.ResponseWasNotFound(existing.Response) {
+			return fmt.Errorf("creating %s: %s was not found", id, routeId)
+		}
+
 		return fmt.Errorf("retrieving existing %s: %+v", *routeId, err)
 	}
 
-	props := routeResp.RouteProperties
-	if props == nil {
-		return fmt.Errorf("%s properties are 'nil': %+v", *routeId, err)
+	customDomains := d.Get("cdn_frontdoor_custom_domain_ids").([]interface{})
+	for _, v := range customDomains {
+		cdId, err := parse.FrontDoorCustomDomainIDInsensitively(v.(string))
```

Comment:
this is coming from the config which is validated and so should be being parsed case-sensitively:

```suggestion
		cdId, err := parse.FrontDoorCustomDomainID(v.(string))
```

Reply:
Fixed.